src/get_imports.rs

```diff
@@ -0,0 +1,244 @@
+use lazy_static::__Deref;
+use std::collections::{HashMap, HashSet};
+use std::iter::FromIterator;
+use std::panic;
+use std::path::{Path, PathBuf};
+use std::rc::Rc;
+use std::sync::Arc;
+use swc_common::errors::{ColorConfig, Handler};
+use swc_common::source_map::Pos;
+use swc_common::SourceFile;
+use swc_ecma_parser::Capturing;
+use swc_ecma_parser::{lexer::Lexer, Parser, StringInput, Syntax};
+
+use crate::error::{GetImportError, GetImportErrorKind};
+
+pub fn get_imports_map_from_file<'a>(
+    file_path: &'a PathBuf,
+) -> Result<HashMap<String, Option<HashSet<String>>>, GetImportError> {
+    get_imports_from_file(&file_path)
+}
+
+fn get_imports_from_file<'a>(
+    file_path: &'a PathBuf,
+) -> Result<HashMap<String, Option<HashSet<String>>>, GetImportError> {
+    let path_string = match file_path.to_str() {
+        Some(path) => path,
+        None => {
+            return Err(GetImportError::new(
+                GetImportErrorKind::ReadImportError,
+                None,
+                None,
+                None,
+            ))
+        }
+    };
+    let cm = Arc::<swc_common::SourceMap>::default();
+    let fm = match cm.load_file(Path::new(path_string)) {
+        Ok(f) => f,
+        Err(e) => {
+            return Err(GetImportError::new(
+                GetImportErrorKind::ReadTsFileError,
+                Some(path_string.to_string()),
+                None,
+                Some(vec![e]),
+            ));
+        }
+    };
+
+    let mut parser_errors: Vec<String> = Vec::new();
+
+    let dest_vector: Vec<u8> = Vec::new();
+
+    let dst = Box::new(dest_vector);
+
+    let handler = Handler::with_emitter_writer(dst, Some(cm.clone()));
+
+    // let handler = Handler::with_tty_emitter(ColorConfig::Auto, true, alse, Some(cm.clone()));
+
+    let lexer = create_lexer(&fm);
+
+    let capturing = Capturing::new(lexer);
+
+    let mut parser = Parser::new_from(capturing);
+
+    let errors = parser.take_errors();
+
+    // panic::set_hook(Box::new(|_| {}));
+
+    if !errors.is_empty() {
+        for error in errors {
+            let mut diagnostic = error.into_diagnostic(&handler);
+            parser_errors.push(diagnostic.message());
+            diagnostic.cancel();
+        }
+        return Err(GetImportError::new(
```

Comment:
if there are complicated initialization requirements here, make different methods for each of the enum members. E.g. `Error::new_parse_err()`, `Error::new_file_read_err()`, etc

Reply:
Got back to the enum approach